### PR TITLE
let redis support sentinel

### PR DESCRIPTION
**Pull request by xmh19936688**
let redis support sentinel

### CODE REVIEW

**Comment by unknwon** [comment]
Thanks for the PR!

I'm not sure familiar with the "Sentinel", can you link a reference?

**Comment by xmh19936688** [comment]
https://github.com/go-macaron/session/issues/26

File: redis/redis.go [hunk 1]
```diff
@@ -154,6 +159,49 @@ func (p *RedisProvider) Init(maxlifetime int64, configs string) (err error) {
...
 
+func (p *RedisProvider) initSentinel(cfg *ini.File) (err error) {
+	opt := &redis.FailoverOptions{}
+
+	for k, v := range cfg.Section("").KeysHash() {
+		switch k {
+		case "master_name":
+			opt.MasterName = v
+		case "sentinel_Addrs":
+			opt.SentinelAddrs = strings.Split(v, "|")
```
Comment (unknwon) [inline redis/redis.go]: Any reason you choose to use `|` over `,` as delimiter here? I feel comma is more nature choice.

**Comment by xmh19936688** [comment]
Comma will be confict with others, here are examples:

single:
```
PROVIDER_CONFIG: addr=1.2.3.4:6379,password=pass
```

sentinel:
```
PROVIDER_CONFIG: ha_mode:sentinel,master_name=mymaster,sentinel_Addrs=1.2.3.4:26379|4.3.2.1:26379,password=pass
```

File: redis/redis.go (hunk 1 above)
Comment (xmh19936688) [reply redis/redis.go]: Comma will be confict with others, here are examples:

single:
```
PROVIDER_CONFIG: addr=1.2.3.4:6379,password=pass
```

sentinel:
```
PROVIDER_CONFIG: ha_mode:sentinel,master_name=mymaster,sentinel_Addrs=1.2.3.4:26379|4.3.2.1:26379,password=pass
```

Comment (unknwon) [reply redis/redis.go]: Gotcha, LGTM then.

**Pull request merged**